features/step_definitions/developer_portal/admin/account/personal_details_steps.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+Given "the buyer wants to edit their personal details" do
+  visit admin_account_personal_details_path
+end
+
+When "the buyer edits their personal details" do
+  fill_in('user[username]', with: 'Alfred')
+  fill_in('user[email]', with: 'alfred@batcave.com')
```

Comment:
IMO, since the form has a new field and it's required, it should be included in the standard "fill form" step. However, the field is not always present so add a check to it. If we want to assert that a certain form has this field I would rather use `I should see field "Current password"` or similar.

```suggestion
  fill_in('user[email]', with: 'alfred@batcave.com')
  fill_in('user[current_password]', with: current_password) if has_css?('user[current_password]')
```